Simple Incremental/Assets/Scripts/EnemyTargeting.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyTargeting : MonoBehaviour
+{
+    // All enemy damage types are added here
+    [SerializeField]
+    List<EnemyDamage> enemyDamage = new List<EnemyDamage>(); 
+
+    // List of gameObjects tagged as player
+    List<GameObject> playerTargets = new List<GameObject>(); 
+
+    // List of player objects and best attack for each
+    List <PlayerTarget> playerTargetList= new List<PlayerTarget>();
+
+    bool targetListCurrent = false;
+    PlayerTarget bestTarget;
+    Vector3[] movement = new Vector3[2];
+
+    // Delay for checking movement to avoid checks on every frame
+    float movementCheckDelay = 1.0f;

```

Comment:
Use a coroutine instead. 

Create a coroutine like this:
while(true)
{
yield return new waitForSeconds(movementCheckDelay)
}

Otherwise you're still updating every frame and wasting CPU!